web/index.js

```diff
@@ -12,12 +11,38 @@ var {Main} = require('./app/books/main')
 var {Gallery} = require('./app/books/gallery')
 var {Book} = require('./app/books/book')
 var {About} = require('./app/pages/about')
+var {Login} = require('./app/pages/login')
+var {Signup} = require('./app/pages/signup')
 
 import {assign} from 'lodash'
+import {UserModel} from './app/model'
+import {updateLocalStorage} from './app/helpers'
 
 class App extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {currentUser: this.props.currentUser}
+  }
+
+  componentDidMount() {
+    UserModel.checkAuth()
+    .then((user) => this.setState({currentUser: user}))
+  }
+
+  setCurrentUser(user) {
+    updateLocalStorage('userToken', user.token)
```

Comment:
The local storage access code. I'd wrap it in something.

On Friday, May 1, 2015, Brandon Martin <notifications@github.com> wrote:

> In web/index.js
> <https://github.com/seanhess/serials/pull/18#discussion_r29543208>:
>
> >
> >  class App extends React.Component {
> > +
> > +  constructor(props) {
> > +    super(props)
> > +    this.state = {currentUser: this.props.currentUser}
> > +  }
> > +
> > +  componentDidMount() {
> > +    UserModel.checkAuth()
> > +    .then((user) => this.setState({currentUser: user}))
> > +  }
> > +
> > +  setCurrentUser(user) {
> > +    updateLocalStorage('userToken', user.token)
>
> Just to clarify did you mean to move the setCurrentUser out to the user
> model or something else?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/seanhess/serials/pull/18/files#r29543208>.
>